src/PanelTraits/Create.php

```diff
@@ -120,4 +144,94 @@ public function syncPivot($model, $data, $form = 'create')
+    {
+        foreach ($formattedData['relations'] as $relationMethod => $relationData) {
+            $model = $relationData['model'];
+            $relation = $item->{$relationMethod}();
+
+            if ($relation instanceof BelongsTo) {
+                $modelInstance = $model::find($relationData['values'])->first();
+                if ($modelInstance != null) {
+                    $relation->associate($modelInstance)->save();
+                } else {
+                    $relation->dissociate()->save();
+                }
+            } elseif ($relation instanceof HasOne) {
+                if ($item->{$relationMethod} != null) {
+                    $item->{$relationMethod}->update($relationData['values']);
+                    $modelInstance = $item->{$relationMethod};
+                } else {
+                    $relationModel = new $model();
+                    $modelInstance = $relationModel->create($relationData['values']);
+                    $relation->save($modelInstance);
+                }
+            } else {
+                $relationModel = new $model();
+                $modelInstance = $relationModel->create($relationData['values']);
+                $relation->save($modelInstance);
+            }
+
+            if (isset($relationData['relations'])) {
+                $this->createRelationsForItem($modelInstance, ['relations' => $relationData['relations']]);
+            }
+        }
+    }
+
+    /**
+     * Get a relation data array from the form data. For each relation defined in the fields through the entity
+     * attribute, set the model, the parent model and the attribute values. For relations defined with the "dot"
+     * notations, this will be used to calculate the depth in the final array (@see \Illuminate\Support\Arr::set() for more).
+     *
+     * @param array $data The form data.
+     * @param string $form Optional form type. Can be either 'create', 'update' or 'both'. Default is 'create'.
+     *
+     * @return array The formatted relation data.
+     */
+    private function getRelationDataFromFormData($data, $form = 'create')
+    {
+        $relationFields = $this->getRelationFields($form);
+
+        $relationData = [];
+        foreach ($relationFields as $relationField) {
+            if (array_key_exists($relationField['name'], $data) && empty($relationField['pivot'])) {
+                $key = implode('.relations.', explode('.', $relationField['entity']));
+                $fieldData = array_get($relationData, 'relations.'.$key, []);
+
+                if (! array_key_exists('model', $fieldData)) {
+                    $fieldData['model'] = $relationField['model'];
+                }
+
+                if (! array_key_exists('parent', $fieldData)) {
+                    $fieldData['parent'] = $this->getRelationModel($relationField['entity'], -1);
+                }
+
+                $fieldData['values'][$relationField['name']] = $data[$relationField['name']];
```

Comment:
Personally I'd life `$relationField['name']` out into a variable for readability but no biggie.